libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -4173,6 +4230,13 @@ reify t f body =
     , toPred $ body x
     ]
 
+explanation :: [String] -> Pred fn -> Pred fn
+explanation _ p@DependsOn {} = p
+explanation _ TruePred = TruePred
+explanation es (FalsePred es') = FalsePred (es ++ es')
+explanation es (Assert es' t) = Assert (es ++ es') t
+explanation es p = Explain es p
+
```

Comment:
So I guess this is where it all comes together. propagating explanations so they don't get lost.

Reply:
Yes, one of many places